internal/wasm/sys.go

```diff
@@ -4,44 +4,14 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"io/fs"
-	"math"
-	"sync/atomic"
 )
 
-// FileEntry maps a path to an open file in a file system.
-//
-// Note: This does not introduce cycles because the types here are in the package "wasi" not "internalwasi".
-type FileEntry struct {
-	Path string
-	FS   fs.FS
-	// File when nil this is a mount like "." or "/".
-	File fs.File
-}
-
 // SysContext holds module-scoped system resources currently only used by internalwasi.
 type SysContext struct {
 	args, environ         []string
 	argsSize, environSize uint32
 	stdin                 io.Reader
 	stdout, stderr        io.Writer
-
```

Comment:
I think maybe there was a misunderstanding. I think we meant to say we usually don't embed like this:

```
type SysContext struct {
*FSContext
}
```

and instead name the field, and only export it if actually used externally.
```
type SysContext struct {
FS *FSContext
}
```

Sorry I think this confusion was caused by me not being clear earlier. I do think that the SysContext should contain an FSContext field. Just if there's one in the go `context.Context` the field value should be ignored for that.

Make sense?

Reply:
oh...I see. It was a relatively big effort honestly😄 I should have asked you to make sure what you meant. I'll revert it then.

Reply:
well good news is I owe you one :D

Reply:
If we don't embed and export `FSContext`, we have to wrap all functions from `FSContext` in `SysContext`. That is exactly why I embedded `FSContext`. For instance,

```
func (c *SysContext) Close(ctx context.Context) error {
	return c.fs.Close(ctx)
}

func (c *SysContext) CloseFile(fd uint32) (bool, error) {
	return c.fs.CloseFile(fd)
}
```

It is applied to `FSConfig` as well. I was feeling like this wrapping is redundant. I hope I get @codefromthecrypt's point this time 😄 I will start defining the same functions in `SysContext`, but please stop me if I'm wrong, @mathetake @codefromthecrypt.

Reply:
This is what it looks like. This commit fails tests, but I want to get your feedback in the early stage.
https://github.com/tetratelabs/wazero/pull/571/commits/21c6d0ced853eb8d2ab0adff1e3a69b545b20d4b

Reply:
Or do you want to add 

```
func (c *SysContext) FS() FSContext {
	return c.fs
}
```

and just use it?

Reply:
I also tried the above approach.
https://github.com/tetratelabs/wazero/pull/571/commits/8e361970044201633d8fbcb6653f11b65462b9c7